lib/delegate.rb

```diff
@@ -343,11 +343,18 @@ def __setobj__(obj)
   end
 end
 
-def Delegator.delegating_block(mid) # :nodoc:
-  lambda do |*args, &block|
-    target = self.__getobj__
-    target.__send__(mid, *args, &block)
-  end.ruby2_keywords
+def Delegator.delegating_block(mid, arity = nil) # :nodoc:
+  if arity != 0
```

Comment:
I was thinking if `def m(**h)` would also have arity 0 but it does not, that's -1 (makes sense, it's 0+ arguments).